meta-balena-common/recipes-containers/resin-supervisor/resin-supervisor/update-resin-supervisor

```diff
@@ -145,9 +145,11 @@ systemctl stop resin-supervisor
 
 # Pull target version.
 echo "Pulling supervisor $image_name:$tag..."
-$DOCKER pull "$image_name:$tag"
-
-$DOCKER rm --force resin_supervisor || true
+if $DOCKER pull "$image_name:$tag"; then
+    $DOCKER rm --force resin_supervisor || true
+else
+    error_handler "supervisor pull failed"
+fi
```

Comment:
@xginn8 There's a second path in the updater script for standalone devices where both the image and the tag are provided on the command line.
In that case, the update should not fail if the pull fails as the image has been provided.
Is this breaking that update path?